### PR TITLE
1. Add RangeLeaderBalancer for inbox store.

### PR DESCRIPTION
**Pull request by mafei6827**
2. Remove some capacities from RecoveryBalancer to ReplicaCntBalancer and make RecoveryBalancer not default active.
3. Fix that InstallSnapshotRequest may not complete correctly when prev InstallSnapshotRequest time out and been cancelled.
4. Fix that TransferLeadership command would not complete correctly when exception occur.

### CODE REVIEW

File: base-kv/base-kv-store-balance-controller/src/main/java/com/baidu/bifromq/basekv/balance/KVRangeBalanceController.java [hunk 1]
```diff
@@ -155,26 +175,25 @@ private void scheduleNow() {
...
                                 metrics.cmdSucceedCounter.increment();
                                 start.stop(metrics.cmdRunTimer);
                                 // Always schedule later after recovery command
                                 if (commandType == CommandType.RECOVERY) {
-                                    scheduling.set(false);
-                                    scheduleLater();
+                                    scheduleLater(randomDelay(), TimeUnit.MILLISECONDS);
                                 } else {
-                                    scheduleNow();
+                                    scheduleLater(1, TimeUnit.SECONDS);
```
Comment (popduke) [inline base-kv/base-kv-store-balance-controller/src/main/java/com/baidu/bifromq/basekv/balance/KVRangeBalanceController.java:189]: why treat recovery as a special case?

File: base-kv/base-kv-store-balance-controller/src/main/java/com/baidu/bifromq/basekv/balance/KVRangeBalanceController.java [hunk 2]
```diff
@@ -155,26 +175,25 @@ private void scheduleNow() {
                     String cmdName = commandToRun.getClass().getSimpleName();
                     Sample start = Timer.start();
                     CommandType commandType = commandToRun.type();
-                    commandRunner.run(commandToRun)
+                    runCommand(commandToRun)
                         .whenCompleteAsync((r, e) -> {
+                            scheduling.set(false);
                             CommandMetrics metrics = metricsManager.getCommandMetrics(balancerName, cmdName);
-                            if (r == CommandRunner.Result.Succeed) {
+                            if (Boolean.TRUE.equals(r)) {
```
Comment (popduke) [inline base-kv/base-kv-store-balance-controller/src/main/java/com/baidu/bifromq/basekv/balance/KVRangeBalanceController.java:182]: why not simplify the condition to just if(r)? r could be null?

File: base-kv/base-kv-store-balance-controller/src/main/java/com/baidu/bifromq/basekv/balance/KVRangeBalanceController.java [hunk 3]
```diff
@@ -187,6 +206,101 @@ private void scheduleNow() {
         scheduling.set(false);
     }
 
+    private CompletableFuture<Boolean> runCommand(BalanceCommand command) {
+        if (command.getExpectedVer() != null) {
+            Long prevCMDVer = historyCommandCache.getIfPresent(command.getKvRangeId());
+            if (prevCMDVer != null && prevCMDVer >= command.getExpectedVer()) {
+                logWarn("[{}]Command version is duplicated with prev one: {}", storeClient.clusterId(), command);
```
Comment (popduke) [inline base-kv/base-kv-store-balance-controller/src/main/java/com/baidu/bifromq/basekv/balance/KVRangeBalanceController.java:213]: clusterId prefix is redundant, log file name contains the information

File: base-kv/base-kv-store-balance-controller/src/main/java/com/baidu/bifromq/basekv/balance/KVRangeBalanceController.java [hunk 4]
```diff
@@ -187,6 +206,101 @@ private void scheduleNow() {
...
+                return CompletableFuture.completedFuture(false);
+            }
+        }
+        logDebug("[{}]Send balanceCommand: {}", storeClient.clusterId(), command);
```
Comment (popduke) [inline base-kv/base-kv-store-balance-controller/src/main/java/com/baidu/bifromq/basekv/balance/KVRangeBalanceController.java:217]: clusterId prefix is redundant, log file name contains the information

File: base-kv/base-kv-store-balance-controller/src/main/java/com/baidu/bifromq/basekv/balance/KVRangeBalanceController.java (hunk 3 above)
Comment (mafei6827) [reply base-kv/base-kv-store-balance-controller/src/main/java/com/baidu/bifromq/basekv/balance/KVRangeBalanceController.java:213]: fixed in next commit

File: base-kv/base-kv-store-balance-controller/src/main/java/com/baidu/bifromq/basekv/balance/KVRangeBalanceController.java (hunk 1 above)
Comment (mafei6827) [reply base-kv/base-kv-store-balance-controller/src/main/java/com/baidu/bifromq/basekv/balance/KVRangeBalanceController.java:189]: The "RecoverRequest" does not have a "version" field, which could lead to it being regenerated and executed redundantly.

File: base-kv/base-kv-store-balance-controller/src/main/java/com/baidu/bifromq/basekv/balance/KVRangeBalanceController.java (hunk 2 above)
Comment (popduke) [reply base-kv/base-kv-store-balance-controller/src/main/java/com/baidu/bifromq/basekv/balance/KVRangeBalanceController.java:182]: like other whenComplete, it
s better to stick to same coding style: check if e != null then handle v

Comment (mafei6827) [reply base-kv/base-kv-store-balance-controller/src/main/java/com/baidu/bifromq/basekv/balance/KVRangeBalanceController.java]: fixed

**Pull request merged**